.buildkite/hooks/pre-command

```diff
@@ -2,14 +2,33 @@
 
 set -euo pipefail
 
-if [[ "$BUILDKITE_PIPELINE_SLUG" == "filebeat" || "$BUILDKITE_PIPELINE_SLUG" == "auditbeat" || "$BUILDKITE_PIPELINE_SLUG" == "heartbeat" ]]; then
+DOCKER_REGISTRY_SECRET_PATH="kv/ci-shared/platform-ingest/docker_registry_prod"
+GITHUB_TOKEN_VAULT_PATH="kv/ci-shared/platform-ingest/github_token"
+
+# Secrets must be redacted
+# https://buildkite.com/docs/pipelines/managing-log-output#redacted-environment-variables
+
+if [[ "$BUILDKITE_PIPELINE_SLUG" == "filebeat" || "$BUILDKITE_PIPELINE_SLUG" == "auditbeat" || "$BUILDKITE_PIPELINE_SLUG" == "heartbeat" || "$BUILDKITE_PIPELINE_SLUG" == "deploy-k8s" ]]; then
   source .buildkite/env-scripts/env.sh
-  source .buildkite/env-scripts/util.sh
   source .buildkite/env-scripts/win-env.sh
+  source .buildkite/env-scripts/util.sh
 
   if [[ -z "${GOLANG_VERSION-""}" ]]; then
     export GOLANG_VERSION=$(cat "${WORKSPACE}/.go-version")
   fi
+
+  if [[ "$BUILDKITE_STEP_KEY" == package* ]]; then
+    export DOCKER_USERNAME_SECRET=$(retry 5 vault kv get -field user "${DOCKER_REGISTRY_SECRET_PATH}")
```

Comment:
warning: the custom vm images are preinstalled with [minfrin/retry](https://github.com/minfrin/retry) -- this is the package that e.g. `apt-get install retry` brings on recent Ubuntu distros.

This binary has a different syntax to the `retry` shell helper we have been using i.e. it'd be `retry -t 5 -- vault kv get ...`.

I understand the `retry` shell function is still part of the library functions under `.buildkite/` but we should watch out for conflicts and at the very least add an issue in https://github.com/elastic/ingest-dev/issues/2885 about consistently using https://github.com/minfrin/retry on Linux post migration.